### DISCUSSION TITLE
Release an Open Source Promise

### DISCUSSION THREAD

**Discussion by sirinath** [modular/modular#1333] [Ideas]
Can Modular publish an open source promise for:
- Mojo and
- Modular Engine

This can include the license details and circumstances on when it will be open sourced.

This will remove the uncertainty and risk around this and help adaptation. Developers can start use the closed source version knowing that it will be available in the future as open source and there will be no vendor lock-in, licensing issues, costs, and unanticipated complication in adaptation.

**Comment by lattner** [comment]
We made a statement on Monday in the keynote: apache2 (LLVM license actually) with the mojo manual now, nightly builds soon, and then mojo stdlib open after that as initial steps.

**Reply by sirinath** [reply]
This is great news! This takes the risk out of being an early adopter of Mojo.

Let me clarify, there is an promise to open source:
- Mojo Docs is open source (LLVM License)
- Mojo Std Lib ((LLVM License)) near future
- Whole of Mojo ((LLVM License)) eventually

Is there a specific trigger which might make the whole of Mojo being opensource? Time, or some measure of maturity.

I watched this on YT but I did not pick up the part of the promise of whole of Mojo.

Still no concrete announcement about Modular Engine being open sourced.
